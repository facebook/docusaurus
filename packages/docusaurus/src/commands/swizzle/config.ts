/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import logger from '@docusaurus/logger';
import {Joi} from '@docusaurus/utils-validation';
import {SwizzleActions, SwizzleActionsStatuses} from './common';
import {getPluginByThemeName} from './themes';
import type {SwizzleComponentConfig, SwizzleConfig} from '@docusaurus/types';
import type {SwizzlePlugin} from './common';

function getModuleSwizzleConfig(
  swizzlePlugin: SwizzlePlugin,
): SwizzleConfig | undefined {
  const getSwizzleConfig =
    swizzlePlugin.plugin.plugin.getSwizzleConfig ??
    swizzlePlugin.plugin.pluginModule?.module.getSwizzleConfig;

  if (getSwizzleConfig) {
    return getSwizzleConfig();
  }

  // TODO deprecate getSwizzleComponentList later
  const getSwizzleComponentList =
    swizzlePlugin.plugin.plugin.getSwizzleComponentList ??
    swizzlePlugin.plugin.pluginModule?.module.getSwizzleComponentList;

  if (getSwizzleComponentList) {
    const safeComponents = getSwizzleComponentList() ?? [];
    const safeComponentConfig: SwizzleComponentConfig = {
      actions: {
        eject: 'safe',
        wrap: 'safe',
      },
      description: undefined,
    };
    return {
      components: Object.fromEntries(
        safeComponents.map((comp) => [comp, safeComponentConfig]),
      ),
    };
  }

  return undefined;
}

const SwizzleConfigSchema = Joi.object<SwizzleConfig>({
  components: Joi.object()
    .pattern(
      Joi.string(),
      Joi.object({
        actions: Joi.object().pattern(
          Joi.string().valid(...SwizzleActions),
          Joi.string().valid(...SwizzleActionsStatuses),
        ),
        description: Joi.string(),
      }),
    )
    .required(),
});

function validateSwizzleConfig(unsafeSwizzleConfig: unknown): SwizzleConfig {
  const result = SwizzleConfigSchema.validate(unsafeSwizzleConfig);
  if (result.error) {
    throw new Error(
      `Swizzle config does not match expected schema: ${result.error.message}`,
    );
  }
  return result.value;
}

export function normalizeSwizzleConfig(
  unsafeSwizzleConfig: unknown,
): SwizzleConfig {
  const swizzleConfig = validateSwizzleConfig(unsafeSwizzleConfig);

  // Ensure all components always declare all actions
  Object.values(swizzleConfig.components).forEach((componentConfig) => {
    SwizzleActions.forEach((action) => {
      if (!componentConfig.actions[action]) {
        componentConfig.actions[action] = 'unsafe';
      }
    });
  });

  return swizzleConfig;
}

const FallbackSwizzleConfig: SwizzleConfig = {
  components: {},
};

export function getThemeSwizzleConfig(
  themeName: string,
  plugins: SwizzlePlugin[],
): SwizzleConfig {
  const plugin = getPluginByThemeName(plugins, themeName);
  const config = getModuleSwizzleConfig(plugin);
  if (config) {
    try {
      return normalizeSwizzleConfig(config);
    } catch (err) {
      logger.error`Invalid Swizzle config for theme name=${themeName}.`;
      throw err;
    }
  }
  return FallbackSwizzleConfig;
}
