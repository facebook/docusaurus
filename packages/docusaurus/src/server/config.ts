/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import fs from 'fs-extra';
import logger from '@docusaurus/logger';
import {
  DEFAULT_CONFIG_FILE_NAME,
  findAsyncSequential,
  loadFreshModule,
} from '@docusaurus/utils';
import {validateConfig} from './configValidation';
import type {LoadContext} from '@docusaurus/types';

async function findConfig(siteDir: string) {
  // We could support .mjs, .ts, etc. in the future
  const candidates = ['.ts', '.mts', '.cts', '.js', '.mjs', '.cjs'].map(
    (ext) => DEFAULT_CONFIG_FILE_NAME + ext,
  );
  const configPath = await findAsyncSequential(
    candidates.map((file) => path.join(siteDir, file)),
    fs.pathExists,
  );
  if (!configPath) {
    logger.error('No config file found.');
    logger.info`Expected one of:${candidates}
You can provide a custom config path with the code=${'--config'} option.`;
    throw new Error();
  }
  return configPath;
}

export async function loadSiteConfig({
  siteDir,
  customConfigFilePath,
}: {
  siteDir: string;
  customConfigFilePath?: string;
}): Promise<Pick<LoadContext, 'siteConfig' | 'siteConfigPath'>> {
  const siteConfigPath = customConfigFilePath
    ? path.resolve(siteDir, customConfigFilePath)
    : await findConfig(siteDir);

  if (!(await fs.pathExists(siteConfigPath))) {
    throw new Error(`Config file at "${siteConfigPath}" not found.`);
  }

  const importedConfig = await loadFreshModule(siteConfigPath);

  const loadedConfig: unknown =
    typeof importedConfig === 'function'
      ? await importedConfig()
      : await importedConfig;

  const siteConfig = validateConfig(
    loadedConfig,
    path.relative(siteDir, siteConfigPath),
  );
  return {siteConfig, siteConfigPath};
}
