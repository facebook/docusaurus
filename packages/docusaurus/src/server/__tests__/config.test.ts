/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import {loadSiteConfig} from '../config';

describe('loadSiteConfig', () => {
  const siteDir = path.join(__dirname, '__fixtures__', 'config');

  it('website with valid siteConfig', async () => {
    const config = await loadSiteConfig({
      siteDir: path.join(__dirname, '__fixtures__', 'simple-site'),
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with ts + js config', async () => {
    const config = await loadSiteConfig({
      siteDir: path.join(
        __dirname,
        '__fixtures__',
        'config/sites/ts-and-js-site',
      ),
    });
    expect(config).toMatchSnapshot();
    // Docusaurus uses in priority a TS config
    expect(config.siteConfig.title).toBe('TS title');
    expect(config).not.toEqual({});
  });

  it('website with .cjs siteConfig', async () => {
    const config = await loadSiteConfig({siteDir});
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid config creator function', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'createConfig.config.js',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid async config', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'configAsync.config.js',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid async config creator function', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'createConfigAsync.config.js',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid JS CJS config', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'configCJS.js',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid JS ESM config', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'configESM.js',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid TypeScript CJS config', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'configCJS.ts',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with valid TypeScript ESM config', async () => {
    const config = await loadSiteConfig({
      siteDir,
      customConfigFilePath: 'configESM.ts',
    });
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  it('website with incomplete siteConfig', async () => {
    await expect(
      loadSiteConfig({
        siteDir,
        customConfigFilePath: 'incomplete.config.js',
      }),
    ).rejects.toThrowErrorMatchingInlineSnapshot(`
            ""url" is required
            "
          `);
  });

  it('website with useless field (wrong field) in siteConfig', async () => {
    await expect(
      loadSiteConfig({
        siteDir,
        customConfigFilePath: 'wrong.config.js',
      }),
    ).rejects.toThrowErrorMatchingInlineSnapshot(`
            "These field(s) ("useLessField",) are not recognized in wrong.config.js.
            If you still want these fields to be in your configuration, put them in the "customFields" field.
            See https://docusaurus.io/docs/api/docusaurus-config/#customfields"
          `);
  });

  it('website with no siteConfig', async () => {
    await expect(
      loadSiteConfig({
        siteDir,
        customConfigFilePath: 'nonExistent.config.js',
      }),
    ).rejects.toThrowErrorMatchingInlineSnapshot(
      `"Config file at "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/nonExistent.config.js" not found."`,
    );
  });
});
