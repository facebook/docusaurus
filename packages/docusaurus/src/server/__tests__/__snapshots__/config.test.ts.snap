// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadSiteConfig website with .cjs siteConfig 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/docusaurus.config.cjs",
}
`;

exports[`loadSiteConfig website with ts + js config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "TS title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/sites/ts-and-js-site/docusaurus.config.ts",
}
`;

exports[`loadSiteConfig website with valid JS CJS config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/configCJS.js",
}
`;

exports[`loadSiteConfig website with valid JS ESM config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/configESM.js",
}
`;

exports[`loadSiteConfig website with valid TypeScript CJS config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/configCJS.ts",
}
`;

exports[`loadSiteConfig website with valid TypeScript ESM config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "plugins": [],
    "presets": [],
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "",
    "themeConfig": {},
    "themes": [],
    "title": "title",
    "titleDelimiter": "|",
    "url": "https://example.com",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/configESM.ts",
}
`;

exports[`loadSiteConfig website with valid async config 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "organizationName": "endiliey",
    "plugins": [],
    "presets": [],
    "projectName": "hello",
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "Hello World",
    "themeConfig": {},
    "themes": [],
    "title": "Hello",
    "titleDelimiter": "|",
    "url": "https://docusaurus.io",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/configAsync.config.js",
}
`;

exports[`loadSiteConfig website with valid async config creator function 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "organizationName": "endiliey",
    "plugins": [],
    "presets": [],
    "projectName": "hello",
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "Hello World",
    "themeConfig": {},
    "themes": [],
    "title": "Hello",
    "titleDelimiter": "|",
    "url": "https://docusaurus.io",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/createConfigAsync.config.js",
}
`;

exports[`loadSiteConfig website with valid config creator function 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [],
    "customFields": {},
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "organizationName": "endiliey",
    "plugins": [],
    "presets": [],
    "projectName": "hello",
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "Hello World",
    "themeConfig": {},
    "themes": [],
    "title": "Hello",
    "titleDelimiter": "|",
    "url": "https://docusaurus.io",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/config/createConfig.config.js",
}
`;

exports[`loadSiteConfig website with valid siteConfig 1`] = `
{
  "siteConfig": {
    "baseUrl": "/",
    "baseUrlIssueBanner": true,
    "clientModules": [
      "foo.js",
    ],
    "customFields": {},
    "favicon": "img/docusaurus.ico",
    "future": {
      "experimental_faster": {
        "swcJsLoader": false,
        "swcJsMinimizer": false,
      },
      "experimental_router": "browser",
      "experimental_storage": {
        "namespace": false,
        "type": "localStorage",
      },
    },
    "headTags": [],
    "i18n": {
      "defaultLocale": "en",
      "localeConfigs": {},
      "locales": [
        "en",
      ],
      "path": "i18n",
    },
    "markdown": {
      "anchors": {
        "maintainCase": false,
      },
      "format": "mdx",
      "mdx1Compat": {
        "admonitions": true,
        "comments": true,
        "headingIds": true,
      },
      "mermaid": false,
      "parseFrontMatter": [Function],
      "preprocessor": undefined,
      "remarkRehypeOptions": undefined,
    },
    "noIndex": false,
    "onBrokenAnchors": "warn",
    "onBrokenLinks": "throw",
    "onBrokenMarkdownLinks": "warn",
    "onDuplicateRoutes": "warn",
    "organizationName": "endiliey",
    "plugins": [
      [
        "@docusaurus/plugin-content-docs",
        {
          "path": "../docs",
        },
      ],
      "@docusaurus/plugin-content-pages",
    ],
    "presets": [],
    "projectName": "hello",
    "scripts": [],
    "staticDirectories": [
      "static",
    ],
    "stylesheets": [],
    "tagline": "Hello World",
    "themeConfig": {},
    "themes": [],
    "title": "Hello",
    "titleDelimiter": "|",
    "url": "https://docusaurus.io",
  },
  "siteConfigPath": "<PROJECT_ROOT>/packages/docusaurus/src/server/__tests__/__fixtures__/simple-site/docusaurus.config.js",
}
`;
